### PR TITLE
Update as per #533

### LINKED ISSUE #533: change Singleton / Pairwise to  AbstractPrior / AbstractRelativeFactor
Or something similar, we can discuss here

# Also see

- https://github.com/JuliaRobotics/IncrementalInference.jl/issues/783

### PR DESCRIPTION
**Pull request by GearsAD**
Closes #533

### CODE REVIEW

**Comment by dehann** [comment]
skipped documntation test, so Travis did pass

File: src/services/DFGFactor.jl [hunk 1]
```diff
@@ -135,7 +135,7 @@ Return `::Bool` on whether given factor `fc::Symbol` is a prior in factor graph
 """
 function isPrior(dfg::G, fc::Symbol)::Bool where G <: AbstractDFG
   fco = getFactor(dfg, fc)
-  getFactorType(fco) isa FunctorSingleton
+  getFactorType(fco) isa AbstractPrior || getFactorType(fco) isa FunctorSingleton
```
Comment (Affie) [inline src/services/DFGFactor.jl:138]: We should just remember to remove this with the deprecation.

Comment (dehann) [reply src/services/DFGFactor.jl:138]: sounds good, the deprecation warnings in DFG v0.9 should be sufficient.  Dropping old abstracts in DFG v0.10 is a good plan

Comment (GearsAD) [reply src/services/DFGFactor.jl:138]: I'm setting up this PR - there are a bunch of places they're still used, we just need to make sure v.10 won't break everything too badly.

**Pull request merged**